### ISSUE TITLE
yield-generator-oscillator crash: "list index out of range"

### ISSUE THREAD

**Issue by dan-da** [JoinMarket-Org/joinmarket#562]
```
2016-06-06 14:43:38,637 [MainThread  ] [DEBUG]  Traceback (most recent call last):
  File "yield-generator-oscillator.py", line 715, in main
    irc.run()
  File "/home/joinmarket/joinmarket/joinmarket/irc.py", line 657, in run
    self.__handle_line(line)
  File "/home/joinmarket/joinmarket/joinmarket/irc.py", line 540, in __handle_line
    self.__handle_privmsg(_chunks[0], _chunks[2], get_irc_text(line))
  File "/home/joinmarket/joinmarket/joinmarket/irc.py", line 491, in __handle_privmsg
    self.__on_privmsg(nick, parsed) 
  File "/home/joinmarket/joinmarket/joinmarket/irc.py", line 369, in __on_privmsg
    self.on_order_fill(nick, oid, amount, taker_pk)
  File "/home/joinmarket/joinmarket/joinmarket/maker.py", line 246, in on_order_fill
    taker_pubkey)
  File "/home/joinmarket/joinmarket/joinmarket/maker.py", line 63, in __init__
    utxo_list, utxo_data = populate_utxo_data()
  File "/home/joinmarket/joinmarket/joinmarket/maker.py", line 52, in populate_utxo_data
    self, oid, amount)
  File "yield-generator-oscillator.py", line 508, in oid_to_order
    lmd = sorted(filtered_mix_balance, key=lambda x: x[1],)[-1]
IndexError: list index out of range 

done
```

**Comment by raedah** [comment]
Please post the output of 'wallet-tool.py yourwal.json summary' if you can.
I am guessing this is an edge case where there is not enough funds in any mixdepth to fulfill the offer.

**Comment by dan-da** [comment]
Your guess appears correct.  The last mix_depth messages before the crash were:

```
2016-06-06 14:43:38,611 [MainThread  ] [DEBUG]  mix depths that have enough = []
2016-06-06 14:43:38,611 [MainThread  ] [DEBUG]  mix depths that have enough with output_size_min, []
```

**Comment by AlexCato** [comment]
PR #551 fixes this for the other yield generators.
I'll fix this for this oscillator later today.

**Comment by AlexCato** [comment]
Fixed in #566 . If you download the new oscillator-file from the development-branch, this shouldnt happen any more.

**Comment by dan-da** [comment]
thx!  trying it now..
